### ISSUE TITLE
[MNG-6946] Build/consumer incorrectly transforms name of artifactId

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8385]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** opened **[MNG-6946](https://issues.apache.org/jira/browse/MNG-6946?redirect=false)** and commented

While trying to build maven-clean-plugin with Maven 3.7.0-SNAPSHOT, most tests called by the maven-invoker-plugins fail.
It looks like the copy of maven-plugins-34.pom is one of the root causes.
Comparing the original and this file (assuming transformation) the following are expected differences:
- removed whitespace around the license
- removed relativePath
- removed whitespaces after closing root-tag

The only invalid difference is
`<artifactId>maven-plugin-annotations</artifactId>`
which became
`<artifactId>tions</artifactId>`

There are 2 issues:
- the artifactId is incorrect
- as this file is not part of the reactor, there's no need to transform the file.

One thing I noticed as well (only happens with build/consumer active):

```
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-bytecode-version) @ maven-clean-plugin ---
ERROR:  'The entity "oslash" was referenced, but not declared.'
```

Workaround: add -Dmaven.experimental.buildconsumer=false

---

**Affects:** 4.0.0

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Root cause found: only first part of this artifactId is available in the buffer, a second call of `characters( char[] ch, int start, int length )` contains the rest.
All Filters need to be rewritten to handle multiple characters-calls correctly.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [a945ca4ed314714fdc2be4616bcf34eaf6d3c1f5](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=a945ca4ed314714fdc2be4616bcf34eaf6d3c1f5)

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

btw, I still see the ERROR.

It seems to be caused by e.g. org.codehaus.plexus:1.0.5:pom, which contains

```xml
<developer>
 <name>Trygve Laugst&oslash;l</name>
 <id>trygvis</id>
 <email>trygvis@codehaus.org</email>
 <roles>
 <role>Developer</role>
 </roles>
 </developer>
```

Need to figure out how to solve this, but that'll be a different ticket.
